test/specs/modules/Dropdown/Dropdown-test.js

```diff
@@ -1980,6 +1980,66 @@ describe('Dropdown', () => {
       wrapper.should.have.state('selectedIndex', 0)
     })
 
+    it('retains the selected item after searching with no results', () => {
+      wrapperMount(<Dropdown options={options} selection search />)
+
+      // open the dropdown
+      wrapper.simulate('click')
+      dropdownMenuIsOpen()
+
+      // select the second item in the list
+      wrapper.find('DropdownItem').at(1).simulate('click')
+      wrapper.find('DropdownItem').at(1).should.have.prop('active', true)
+      wrapper.find('DropdownItem').at(1).should.have.prop('selected', true)
+
+      wrapper.find('DropdownItem').at(0).should.have.prop('active', false)
+      wrapper.find('DropdownItem').at(0).should.have.prop('selected', false)
+
+      // search for a non-existent item, triggering the not found message
+      const search = wrapper.find('input.search')
+      search.simulate('change', { target: { value: '__nonExistingSearchQuery__' } })
+
+      // click outside, close the dropdown
+      domEvent.click(document)
```

Comment:
This line isn't working really.  I noted that once you added describe.only, the call domEvent.click(document) isn't really working, and would also cause a failure in the test `closes on click outside` because it uses the same call.  If you use `document.body.click()` (like the test `blurs the Dropdown node on close by clicking outside component`) or `domEvent.click(document.body)` then things start working.  What this causes though is a call to `closeOnDocumentClick`, which causes the dropdown menu to close, but the focus is still on the input and the input search text is never cleared.   In this case, this is correct behavior, since the menu has been closed, but focus is still on the input, so the invalid search text should not be removed.  What the test should be doing is calling the `handleBlur` function which is what you're trying to test with this test case.  In summary,
1. Your checks below with the wrapper.find() calls aren't working because the dropdown still has the search text in the input and is showing the No results message.
2. Need a different way to trigger a blur event instead of document.body.click().  This applies to this test case and to the `blurs the Dropdown node on close by clicking outside component` case.  May need to include another button element to throw the focus to cause a blur to happen.  
3. Actually, I believe the dropdown is working well with your changes, its just the test case isn't correctly testing it